server/src/frontend_api.rs

```diff
@@ -568,10 +569,41 @@ security(
 )]
 #[post("/client/metrics")]
 async fn post_frontend_metrics(
+    req: HttpRequest,
     edge_token: EdgeToken,
+    connect_via: Data<ConnectVia>,
     metrics: Json<ClientMetrics>,
     metrics_cache: Data<MetricsCache>,
 ) -> EdgeResult<HttpResponse> {
+    if let Some(version) = req
+        .headers()
+        .get("unleash-sdk")
+        .and_then(|val| val.to_str().ok())
+        .map(str::to_owned)
+    {
```

Comment:
We might want to move this into an extractor, so you can just put it as an argument to the handler.
With an extractor, you make your self a UnleashSdkHeader type, and then write an extractor that explains how to go from a request to an UnleashSdkHeader.

Then you can simply have 
`sdk_version: UnleashSdkHeader` as an argument to the handler method.

Reply:
Extracted